File: backend/app/Models/ProductionOrder.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class ProductionOrder extends Model
{
    use HasFactory;

    protected $fillable = [
        'order_id', 'stock_id', 'required_meter',
        'production_line', 'start_date', 'end_date',
        'progress_percentage', 'status', 'notes',
    ];

    protected $casts = [
        'start_date' => 'date',
        'end_date' => 'date',
        'progress_percentage' => 'integer',
    ];

    public function order()
    {
        return $this->belongsTo(Order::class);
    }

    public function stock()
    {
        return $this->belongsTo(Stock::class);
    }

    public function stockMovements()
    {
        return $this->hasMany(StockMovement::class);
    }

    public function productionLogs()
    {
        return $this->hasMany(ProductionLog::class)->orderBy('date')->orderBy('shift');
    }

    // Tüm loglardan toplam üretilen adet
    public function getTotalProducedAttribute(): int
    {
        return $this->productionLogs()->sum('produced_quantity');
    }

    public function qualityControls()
    {
        return $this->hasMany(QualityControl::class);
    }
}

File: backend-dotnet/Models/ProductionOrder.cs
namespace TexTrack.Api.Models;

public class ProductionOrder
{
    public long Id { get; set; }
    public long OrderId { get; set; }
    public long? StockId { get; set; }
    public decimal? RequiredMeter { get; set; }
    public string ProductionLine { get; set; } = string.Empty;
    public DateOnly? StartDate { get; set; }
    public DateOnly? EndDate { get; set; }
    public int ProgressPercentage { get; set; }
    public string Status { get; set; } = "Waiting";
    public string? Notes { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }

    public Order? Order { get; set; }
    public Stock? Stock { get; set; }
    public ICollection<StockMovement> StockMovements { get; set; } = new List<StockMovement>();
    public ICollection<ProductionLog> ProductionLogs { get; set; } = new List<ProductionLog>();
    public ICollection<QualityControl> QualityControls { get; set; } = new List<QualityControl>();
}
